src/main/java/com/atherys/towns/facade/TownRaidFacade.java

```diff
@@ -0,0 +1,271 @@
+package com.atherys.towns.facade;
+
+import com.atherys.core.economy.Economy;
+import com.atherys.towns.AtherysTowns;
+import com.atherys.towns.TownsConfig;
+import com.atherys.towns.api.command.TownsCommandException;
+import com.atherys.towns.model.RaidPoint;
+import com.atherys.towns.model.entity.Resident;
+import com.atherys.towns.model.entity.Town;
+import com.atherys.towns.service.PlotService;
+import com.atherys.towns.service.ResidentService;
+import com.atherys.towns.service.TownRaidService;
+import com.atherys.towns.util.MathUtils;
+import com.flowpowered.math.vector.Vector3d;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.slf4j.Logger;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.effect.particle.ParticleType;
+import org.spongepowered.api.effect.particle.ParticleTypes;
+import org.spongepowered.api.entity.AreaEffectCloud;
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.entity.EntityTypes;
+import org.spongepowered.api.entity.Transform;
+import org.spongepowered.api.entity.living.player.Player;
+import org.spongepowered.api.event.cause.Cause;
+import org.spongepowered.api.event.entity.DestructEntityEvent;
+import org.spongepowered.api.event.entity.living.humanoid.player.RespawnPlayerEvent;
+import org.spongepowered.api.service.economy.account.Account;
+import org.spongepowered.api.text.Text;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.math.BigDecimal;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.*;
+
+import static org.spongepowered.api.text.format.TextColors.*;
+
+@Singleton
+public class TownRaidFacade {
+
+    @Inject
+    TownsConfig config;
+
+    @Inject
+    TownFacade townFacade;
+
+    @Inject
+    TownsMessagingFacade townsMsg;
+
+    @Inject
+    TownRaidService townRaidService;
+
+    @Inject
+    ResidentService residentService;
+
+    @Inject
+    PlotService plotService;
+
+    public TownRaidFacade() {
+
+    }
+
+    public boolean isEntityRaidPoint(Entity entity) {
+        return townRaidService.isIdRaidEntity(entity.getUniqueId());
+    }
+
+    public boolean isTownTooClose(Town town, Location<World> spawnLocation) {
+        Logger logger = AtherysTowns.getInstance().getLogger();
+        double test = MathUtils.getSmallestDistanceToTown(town.getPlots(), MathUtils.vec3dToVec2i(spawnLocation.getPosition()));
```

Comment:
I think the question we want to ask here is if any plot is within a distance, not what is the smallest distance to a set of plots. Difference between these two is that we don't need to check every plot.

I still think we need to check plots within a radius of the raid point, not plots of the target town, as players could possibly exploit this by targeting a different town to get a closer raid point to a town?